bok_choy/a11y/axe_core_ruleset.py

```diff
@@ -0,0 +1,296 @@
+"""
+Interface for using the axe-core ruleset in A11yAudit.
+See: https://github.com/dequelabs/axe-core
+"""
+import json
+import os
+import requests
+
+from pprint import pformat
+from textwrap import dedent
+
+from .helpers import AccessibilityError
+from ..promise import Promise
+
+
+CUR_DIR = os.path.dirname(os.path.abspath(__file__))
+
+
+class AxeCoreAuditConfig(object):
```

Comment:
An example in this codebase is the PageObject metaclass that page objects inherit from. Defined page objects override the abstract methods like is_browser_on_page, url, etc. 
Here you would make metaclasses for A11yAuditConfig and A11yAudit, with abstract methods that return NotImplementedError. That gives placeholders for the "plug-ins", in your case the Axs and AxeCore to override. And when a third ruleset gets added in it will follow the same pattern.
Hope this makes sense. We can discuss in person if it's more expedient.